lib/src/ruby/ruby.cc

```diff
@@ -55,6 +58,69 @@ namespace facter { namespace ruby {
          load_custom_facts(facts, false, paths);
     }
 
+    value const* lookup(value const* value, vector<string>::iterator segment, vector<string>::iterator end) {
+        auto rb_value = dynamic_cast<ruby_value const*>(value);
+        if (!rb_value) {
+             return nullptr;
+        }
+
+        // Check for a cached lookup
+        auto key = accumulate(segment, end, string{}, [](const string& a, const string& b) -> string {
```

Comment:
Oh, there's a bug here. If a hash has a key called "foo.bar" as well as a key called "foo" that has a hash with a key "bar", this won't work. I need to wrap in quotes when there are dots in a string (just like the actual query string would have)